src/main/java/duchess/Duchess.java

```diff
@@ -0,0 +1,226 @@
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
+
+
+    /**
+     * Marks a task as completed.
+     *
+     * @param in The user input specifying the task number.
+     * @throws DuchessException If the input format is invalid.
+     */
+    public void mark(String in) throws DuchessException {
+        int taskNum;
+        try {
+            String taskNumStr = in.split(" ")[1];
+            taskNum = Integer.parseInt(taskNumStr);
+            this.taskList.get(taskNum - 1).mark();
+            this.ui.showItemMarked();
+        } catch (Exception e) {
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
+
+    /**
+     * Unmarks a task as not completed.
+     *
+     * @param in The user input specifying the task number.
+     * @throws DuchessException If the input format is invalid.
+     */
+    public void unmark(String in) throws DuchessException {
+        int taskNum;
+        try {
+            String taskNumStr = in.split(" ")[1];
+            taskNum = Integer.parseInt(taskNumStr);
+            this.taskList.get(taskNum - 1).unmark();
+            this.ui.showItemUnmarked();
+        } catch (Exception e) {
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
+
+    /**
+     * Deletes a task from the task list.
+     *
+     * @param in The user input specifying the task number.
+     * @throws DuchessException If the input format is invalid.
+     */
+    public void deleteTask(String in) throws DuchessException {
+        int taskNum;
+        try {
+            String taskNumStr = in.split(" ")[1];
+            taskNum = Integer.parseInt(taskNumStr);
+            this.taskList.remove(taskNum - 1);
+            this.ui.showItemDeleted();
+        } catch (Exception e) {
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
+    /**
+     * Handles unrecognized commands by throwing an exception.
+     *
+     * @param command The unrecognized command input by the user.
+     * @throws DuchessException Always thrown with an invalid command error.
+     */
+    public void processUnrecognisedCommand(String command) throws DuchessException {
+        throw new DuchessException(command, ErrorType.INVALID_COMMAND);
+    }
+
+    /**
+     * Finds items in the task list that match the keyword and displays it.
+     *
+     * @param in The command input by the user.
+     */
+    public void find(String in) {
+        String keyword = in.substring(in.indexOf(" ") + 1);
+        ArrayList<Task> matchingTasks = this.taskList.find(keyword);
+        this.ui.showMatchingTasks(matchingTasks); 
+    }
+
+    /**
+     * Starts the Duchess application, handling user input in a loop.
+     */
+    public void start(){
+        while (isRunning) {
+            try {
+                in = this.ui.readCommand();
+                String[] commandStr= this.parser.parseCommand(in);
```

Comment:
May have missed a space here.

```suggestion
                String[] commandStr = this.parser.parseCommand(in);
```